### PR TITLE
Feat/oidc client

### PR DESCRIPTION
**Pull request by OzgeYurtsever**

### CODE REVIEW

File: public/index.html [hunk 1]
```diff
@@ -41,6 +43,18 @@
       To begin the development, run `npm start` or `yarn start`.
       To create a production bundle, use `npm run build` or `yarn build`.
     -->
-</body>
-
-</html>
\ No newline at end of file
+  </body>
+  <script src="oidc-client.min.js"></script>
```
Comment (emelalkim) [inline public/index.html:47]: add %PUBLIC_URL% in the beginning

File: src/App.js [hunk 2]
```diff
@@ -44,6 +43,8 @@ import 'react-toastify/dist/ReactToastify.css';
 import './App.css';
 import RightsideBar from './components/RightsideBar/RightsideBar';
 import MinimizedReport from './components/searchView/MinimizedReport';
+import { AuthConsumer } from 'providers/authProvider';
```
Comment (emelalkim) [inline src/App.js:46]: check if used

File: src/App.js [hunk 3]
```diff
@@ -69,10 +70,15 @@ const reportsList = [
 class App extends Component {
   constructor(props) {
     super(props);
+    this.authority = 'http://bds-c02xf0r0jhd5.local:8899/auth/realms/ePad';
```
Comment (emelalkim) [inline src/App.js:73]: clean up

File: src/App.js [hunk 4]
```diff
@@ -98,7 +104,12 @@ class App extends Component {
       reportsCompArr: [],
       minReportsArr: [],
       hiddenReports: {},
-      metric: null
+      metric: null,
```
Comment (emelalkim) [inline src/App.js:107]: is it being used?

File: src/App.js [hunk 5]
```diff
@@ -507,36 +518,62 @@ class App extends Component {
 
   async componentDidMount() {
     localStorage.setItem('treeData', JSON.stringify({}));
+    console.log('process.env.PUBLIC_URL', process.env.PUBLIC_URL);
```
Comment (emelalkim) [inline src/App.js:521]: clean up

File: src/App.js [hunk 6]
```diff
@@ -507,36 +518,62 @@ class App extends Component {
...
     Promise.all([
       fetch(`${process.env.PUBLIC_URL}/config.json`),
-      fetch(`${process.env.PUBLIC_URL}/keycloak.json`)
     ])
       .then(async results => {
+        console.log(' in then top 1');
```
Comment (emelalkim) [inline src/App.js:526]: clean up

File: src/App.js [hunk 7]
```diff
@@ -507,36 +518,62 @@ class App extends Component {
...
+        sessionStorage.setItem('response_type', response_type);
+        sessionStorage.setItem('scope', scope);
+        sessionStorage.setItem('redirected', true);
+
+        const clientId = client_id;
+        const redirectUri = redirect_uri;
+        const responseType = response_type;
+
+        console.log(' in then after set 2');
+        this.setState({
```
Comment (emelalkim) [inline src/App.js:565]: no state

File: src/App.js [hunk 8]
```diff
@@ -574,92 +611,48 @@ class App extends Component {
...
-  completeAutorization = apiUrl => {
-    let getAuthUser = null;
-
-    if (sessionStorage.getItem('authMode') !== 'external') {
-      const keycloak = Keycloak(
-        JSON.parse(sessionStorage.getItem('keycloakJson'))
+  completeAutorization = async apiUrl => {
+    this.authService = new auth.AuthService();
+    const authenticated = this.authService.isAuthenticated();
+    if (authenticated) {
```
Comment (emelalkim) [inline src/App.js:617]: consider moving redirect here

File: src/App.js [hunk 9]
```diff
@@ -574,92 +611,48 @@ class App extends Component {
...
-      .catch(err2 => {
-        console.log('Authentication failed!', err2);
-      });
+  registerUser = async (userInfo, authService, authenticated, apiUrl) => {
+    const source = userInfo.profile ? userInfo.profile : userInfo;
+    let user = {
+      user: source.preferred_username || source.email,
+      displayname: source.given_name
+    };
+    // TODO: implement login with authService
```
Comment (emelalkim) [inline src/App.js:634]: remove todo

File: src/App.js [hunk 10]
```diff
@@ -732,14 +726,12 @@ class App extends Component {
...
       user: null
     });
-    if (sessionStorage.getItem('authMode') !== 'external')
-      this.state.keycloak.logout().then(() => {
-        this.setState({
-          keycloak: null
-        });
-        auth.logout();
+    if (sessionStorage.getItem('authMode') !== 'external') {
+      this.authService.logout();
```
Comment (emelalkim) [inline src/App.js:730]: this is already called. do you need another one?

File: src/App.js [hunk 11]
```diff
@@ -964,6 +956,23 @@ class App extends Component {
         return all;
       }, 0);
     }
+
+    const {
```
Comment (emelalkim) [inline src/App.js:960]: check if we use this state

File: src/App.js [hunk 12]
```diff
@@ -1047,6 +1065,16 @@ class App extends Component {
             >
               <Switch className="splitted-mainview">
                 <Route path="/logout" component={Logout} />
+                <Route
```
Comment (emelalkim) [inline src/App.js:1068]: is this used? remove if not?

File: src/App.js [hunk 13]
```diff
@@ -1032,10 +1041,19 @@ class App extends Component {
...
         {this.state.minReportsArr}
-        {!this.state.authenticated && mode !== 'lite' && (
-          <Route path="/login" component={LoginForm} />
+        {!this.state.authenticated && (
+          <Route
+            path="/login"
+            render={props => (
+              <LoginForm
+                {...props}
+                authService={this.authService}
```
Comment (emelalkim) [inline src/App.js:1050]: are they used?

File: src/App.js [hunk 14]
```diff
@@ -1160,8 +1187,6 @@ class App extends Component {
                     />
                   )}
                 />
-
-                <Redirect to="/not-found" />
```
Comment (emelalkim) [inline src/App.js:1164]: do you want to remove or keep?

File: src/Utils/authConst.js [hunk 15]
```diff
@@ -0,0 +1,167 @@
+export const IDENTITY_CONFIG = {
```
Comment (emelalkim) [inline src/Utils/authConst.js:1]: file not required anymore, delete

File: src/components/loginForm.jsx [hunk 16]
```diff
@@ -1,39 +1,46 @@
...
+      authService: null
     };
   }
 
+  componentDidMount = () => {
+    const authService = new auth.AuthService();
+    this.setState({ authService });
+    authService.signinRedirect({});
+
+    // sessionStorage.setItem('authService', JSON.stringify(authService));
```
Comment (emelalkim) [inline src/components/loginForm.jsx:23]: clean up

File: src/components/sideBar/sidebar.jsx [hunk 17]
```diff
@@ -104,12 +107,12 @@ class Sidebar extends Component {
 
         projects = projects.concat(all, nonassigned);
 
-        // const pid = projects[0].id;
-        // this.setState({ projects, pid, selected: pid });
-        // if (this.props.openSeries.length === 0) {
-        //   this.props.history.push(`/search/${pid}`);
-        // }
-        // this.props.getPidUpdate(pid);
+        const pid = projects[0].id;
```
Comment (emelalkim) [inline src/components/sideBar/sidebar.jsx:110]: check this out

File: src/providers/authProvider.jsx [hunk 18]
```diff
@@ -0,0 +1,25 @@
+import React, {Component} from "react";
```
Comment (emelalkim) [inline src/providers/authProvider.jsx:1]: we can remove this file

File: src/services/authService.js [hunk 19]
```diff
@@ -1,74 +1,239 @@
...
+
+const apiUrl = sessionStorage.getItem('apiUrl');
+const mode = sessionStorage.getItem('mode');
 // we need the keycloak object to be able to use update token
-let keycloak = null;
+
+// let keycloak = null;
+let authService = null;
 
 export function refreshToken(keycloak, minValidity) {
```
Comment (emelalkim) [inline src/services/authService.js:14]: we do not need this anymore

File: src/services/authService.js [hunk 20]
```diff
@@ -1,74 +1,239 @@
...
-  sessionStorage.removeItem("lastSavedAim");
+  sessionStorage.removeItem('lastSavedAim');
 }
 
 export function getCurrentUser() {
-  return sessionStorage.getItem("username");
+  return sessionStorage.getItem('username');
 }
 
+// export async function getAuthHeader() {
```
Comment (emelalkim) [inline src/services/authService.js:53]: clean up

File: src/services/authService.js [hunk 21]
```diff
@@ -1,74 +1,239 @@
...
 export async function getAuthHeader() {
-  if (this.keycloak) {
-    try {
-      await refreshToken(this.keycloak, 5);
-      const header = `Bearer ${this.keycloak.token}`;
+  authService = new AuthService();
+  try {
+    const user = await authService.getUser();
+    if (user.access_token) {
+      // TODO: refresh token
```
Comment (emelalkim) [inline src/services/authService.js:78]: clean up todo

File: public/index.html (hunk 1 above)
Comment (OzgeYurtsever) [reply public/index.html:47]: done!

File: src/App.js (hunk 2 above)
Comment (OzgeYurtsever) [reply src/App.js:46]: deleted

File: src/App.js (hunk 3 above)
Comment (OzgeYurtsever) [reply src/App.js:73]: done!

File: src/App.js (hunk 4 above)
Comment (OzgeYurtsever) [reply src/App.js:107]: For another feature! only styling change here

File: src/App.js (hunk 5 above)
Comment (OzgeYurtsever) [reply src/App.js:521]: done

File: src/App.js (hunk 6 above)
Comment (OzgeYurtsever) [reply src/App.js:526]: done

File: src/App.js (hunk 7 above)
Comment (OzgeYurtsever) [reply src/App.js:565]: done

File: src/services/authService.js (hunk 21 above)
Comment (OzgeYurtsever) [reply src/services/authService.js:78]: done

File: src/services/authService.js (hunk 20 above)
Comment (OzgeYurtsever) [reply src/services/authService.js:53]: done!

File: src/Utils/authConst.js (hunk 15 above)
Comment (OzgeYurtsever) [reply src/Utils/authConst.js:1]: done!

File: src/App.js (hunk 9 above)
Comment (OzgeYurtsever) [reply src/App.js:634]: done!

File: src/App.js (hunk 10 above)
Comment (OzgeYurtsever) [reply src/App.js:730]: done!

File: src/App.js (hunk 11 above)
Comment (OzgeYurtsever) [reply src/App.js:960]: cleaned!

File: src/App.js (hunk 12 above)
Comment (OzgeYurtsever) [reply src/App.js:1068]: deleted

File: src/App.js (hunk 13 above)
Comment (OzgeYurtsever) [reply src/App.js:1050]: Will keep!

File: src/App.js (hunk 14 above)
Comment (OzgeYurtsever) [reply src/App.js:1164]: will keep!

File: src/components/loginForm.jsx (hunk 16 above)
Comment (OzgeYurtsever) [reply src/components/loginForm.jsx:23]: done

File: src/providers/authProvider.jsx (hunk 18 above)
Comment (OzgeYurtsever) [reply src/providers/authProvider.jsx:1]: file deleted

File: src/services/authService.js (hunk 19 above)
Comment (OzgeYurtsever) [reply src/services/authService.js:14]: done!

**Pull request merged**